### PR TITLE
Chapter 13: User microposts (Part 3)

### PR DESCRIPTION
**Pull request by lyvanminh**

### CODE REVIEW

**Comment by lyvanminh** [comment]
screenshot from 2018-09-14 13-57-39

**Comment by lyvanminh** [comment]
ready

File: app/controllers/microposts_controller.rb [hunk 1]
```diff
@@ -14,11 +15,20 @@ def create
...
 
   private
 
   def micropost_params
-    params.require(:micropost).permit :content
+    params.require(:micropost).permit :content, :picture
+  end
+
+  def correct_user
+    @micropost = current_user.microposts.find_by(id: params[:id])
```
Comment (thanhhungchu95) [inline app/controllers/microposts_controller.rb]: bỏ ()

File: app/controllers/microposts_controller.rb [hunk 2]
```diff
@@ -14,11 +15,20 @@ def create
...
+    redirect_to root_url if @micropost.nil?
```
Comment (thanhhungchu95) [inline app/controllers/microposts_controller.rb]: unless @micropost

File: app/models/micropost.rb [hunk 3]
```diff
@@ -1,6 +1,15 @@
...
   scope :created_at, ->{order created_at: :desc}
+  mount_uploader :picture, PictureUploader
   validates :content, presence: true,
     length: {maximum: Settings.models.micropost.content_maximum}
+  validate  :picture_size
+
+  private
+
+  def picture_size
+    return unless picture.size > 1.megabytes
```
Comment (thanhhungchu95) [inline app/models/micropost.rb]: đưa 1 vào config

File: app/views/shared/_micropost_form.html.erb [hunk 4]
```diff
@@ -1,7 +1,19 @@
...
+  <div class="field">
     <%= f.text_area :content, placeholder: t(".compose_new") %>
   </div>
   <%= f.submit t(".post"), class: "btn btn-primary" %>
+  <span class="picture">
+  <%= f.file_field :picture, accept: "image/jpeg,image/gif,image/png" %>
+  </span>
 <% end %>
+
+<script type="text/javascript">
```
Comment (thanhhungchu95) [inline app/views/shared/_micropost_form.html.erb]: đưa script vào file riêng

**Comment by thanhhungchu95** [comment]
commented

**Pull request merged**